src/main/java/core/basesyntax/MyLinkedList.java

```
@@ -1,50 +1,154 @@
+
 package core.basesyntax;
 
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int count;
+
+    public MyLinkedList() {
+        count = 0;
+    }
+
+    private class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(T item, Node<T> next, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> temp = new Node<>(value, null, lastNode);
+        if (count == 0) {
+            firstNode = temp;
+        } else {
+            lastNode.next = temp;
+        }
+        lastNode = temp;
+        count++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= count) {
```

let's throw an exception and then do adding